### PR TITLE
move stage particle boundaries to compile unit

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- [ ] PR #5090 is combining this and other PRs to reduce the CI load

### CODE REVIEW

File: include/picongpu/simulation/stage/ParticleBoundaries.hpp [hunk 1]
```diff
@@ -212,26 +40,13 @@ namespace picongpu
...
-                {
-                    processSpecies();
-                }
-
-            private:
-                //! Only allow customization for species with pusher
-                using SpeciesWithPusher =
-                    typename pmacc::particles::traits::FilterByFlag<VectorAllSpecies, particlePusher<>>::type;
-                //! Functor to process all affected species
-                meta::ForEach<SpeciesWithPusher, detail::ParticleBoundariesCommandLine<boost::mpl::_1>> processSpecies;
```
Comment (ikbuibui) [inline include/picongpu/simulation/stage/ParticleBoundaries.hpp:234]: Just for information, why was this moved into both the functions rather than being left here

Comment (psychocoderHPC) [reply include/picongpu/simulation/stage/ParticleBoundaries.hpp:234]: This avoids that this compile-time algorithm is executed even if it is not required.
The evaluation is now in the cpp because it is only there required.
This reduces the compile time.

**Review by ikbuibui — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
merged together with https://github.com/ComputationalRadiationPhysics/picongpu/pull/5090 due to rebases of https://github.com/ComputationalRadiationPhysics/picongpu/pull/5090 in between github was not able to match this PR and auto close it.
